### PR TITLE
Update OWA GEP to better reflect current state

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**How to categorize this PR?**

/area documentation
/kind cleanup

**What this PR does / why we need it**:
This PR updates the OWA GEP to better reflect the current state of the project.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Typically, we don't maintain the GEPs after implementation but rather add dedicated documents for the respective features. I'm fine with merging this, though.

**Comment by dimityrmirchev** [comment]
Ok, thanks! I just wanted to clarify/remove some things from the GEP that were never implemented so that no one reading it gets confused.

File: docs/proposals/12-oidc-webhook-authenticator.md [hunk 1]
```diff
@@ -96,33 +96,20 @@ This new OWA can be configured with multiple OIDC providers and the entire flow
...
-      name: foo
-    spec:
-      issuerURL: https://foo.bar
-      ...
-    status:
-      keys: f31deA9b... #the content of jwks_uri base64-encoded
-    ```
-
-1. OWA uses those keys, issuer, client_id and other settings to add OIDC authenticator to a in-memory list of [Token Authenticators](https://pkg.go.dev/k8s.io/apiserver/pkg/authentication/authenticator?tab=doc#Token).
+1. OWA watches for changes on this resource and does [OIDC discovery](https://openid.net/specs/openid-connect-discovery-1_0.html). The [OIDC provider's  configuration](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse) has to be accessible under the `spec.issuerURL` with a [well-known path (.well-known/openid-configuration)](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig).
```
Comment (plkokanov) [inline docs/proposals/12-oidc-webhook-authenticator.md]:
```suggestion
1. OWA watches for changes on this resource and does [OIDC discovery](https://openid.net/specs/openid-connect-discovery-1_0.html). The [OIDC provider's configuration](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse) has to be accessible under the `spec.issuerURL` with a [well-known path (.well-known/openid-configuration)](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig).
```

**Review by plkokanov — CHANGES_REQUESTED** [review]
Just one very very very small nit.

**Review by rfranzke — COMMENTED** [review]
/approve

**Review by rfranzke — COMMENTED** [review]
As this is only documentation, let's get it in
/milestone v1.45
/lgtm

**Pull request merged**